pkgs/applications/science/logic/surelog/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, python3
+, pkg-config
+, libuuid
+, openjdk11
+, gperftools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "Surelog";
+  version = "2022.05.15";
+
+  src = fetchFromGitHub {
+    owner = "chipsalliance";
+    repo = pname;
+    rev = "15d3698ca5c7d45dd95b58c15e76131420cb001c";
+    hash = "sha256-dfje9yZ8ZR7x1EUxDUpKNcOWKYTPwPG6T4HzudV59R4=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    openjdk11
+    (python3.withPackages (p: with p; [
+      psutil
+      orderedmultidict
+    ]))
+  ];
+
+  buildInputs = [
+    libuuid
+    gperftools
+  ];
+
+  doCheck = true;
+  checkPhase = ''
+    runHook preCheck
+    make -j $NIX_BUILD_CORES UnitTests
+    ctest --output-on-failure
+    runHook postCheck
+  '';
+
+  postInstall = ''
+    mv $out/lib/surelog/* $out/lib/
+    rm -rf $out/lib/surelog
+    rm -rf $out/lib/pkg
```

Comment:
mmh, I mean that we should _keep_ the pkg, in lib/surelog, because I think surelog is expecting it there (though I am not entirely sure if it will find it on Nixos). It is its private set of pre-compiled bulit-ins that it needs to access in certain contexts. I'd probably keep it there for now and we can worry later if that needs to be adapted.

(Also, do we need to move the libraries from `lib/surelog` to `lib` ? I suspect that is so that we can uniformly use it in Nix, so more a question of understanding than critique.
If we do, do we get into trouble for surelog also copying the libflatbuffers and antlr runtime it creates ? I'd probably be more comfortable if they were all together in one private surelog namespace, but I suppose Nix provides us that namespace by definition.
(sorty, Nix noob here))

Reply:
I _think_ I moved those libraries because they weren't being found otherwise, but I don't honestly remember. I can look into this a bit more closely, but it might take me a few days before I can get to it. In the meantime, I moved pkg back into lib/surelog.